core/heap.c

```diff
@@ -1901,12 +1901,20 @@ vmm_heap_init()
     if (DYNAMO_OPTION(vm_reserve)) {
         vmm_heap_unit_init(&heapmgt->vmcode, DYNAMO_OPTION(vm_size), true, "vmcode");
         if (!REACHABLE_HEAP()) {
-            vmm_heap_unit_init(&heapmgt->vmheap, DYNAMO_OPTION(vmheap_size), false,
-                               "vmheap");
+            vmm_heap_unit_init(
+                &heapmgt->vmheap,
+#ifdef WINDOWS
+                (DYNAMO_OPTION(inject_x64) && is_wow64_process(NT_CURRENT_PROCESS))
```

Comment:
Probably safer to remove the inject_x64 check and have IF_X64 instead since we don't care about injection options, just whether we're a 64-bit DR in a WOW64 process.

IF_WINDOWS_ELSE and IF_X64_ELSE could be used to avoid whole-line ifdefs.